integration/test_client.py

```diff
@@ -539,3 +539,21 @@ def test_client_is_ready() -> None:
     assert weaviate.connect_to_wcs(
         cluster_url=WCS_URL, auth_credentials=WCS_CREDS, skip_init_checks=True
     ).is_ready()
+
+
+def test_local_proxies() -> None:
+    with weaviate.connect_to_local(
+        additional_config=wvc.init.AdditionalConfig(
+            proxies={
+                "http": "http://localhost:8075",
+                "grpc": "http://localhost:10000",
```

Comment:
Yeh, I thought about it but do we also want to support a free-form key input such as `http://localhost:8080/v1/objects: http://localhost:5000`? It seems amazingly niche and pretty pointless but something to consider nonetheless

Actually, we could do it like the new `Timeout` class and have a union. I'll implement that

Reply:
> Yeh, I thought about it but do we also want to support a free-form key input such as http://localhost:8080/v1/objects: http://localhost:5000? It seems amazingly niche and pretty pointless but something to consider nonetheless

What would be the point of that? To only proxy calls to specific endpoints?

Reply:
Exactly that, pretty niche and borderline pointless

Reply:
so should we even support it? If someone comes with a good usecase we could still add the dict back

Reply:
I think the `dict` should stay for BC and ease of migration but we can communicate the class too

Reply:
but we don't need to keep BC because it just didn't work before? And migration frm v3is not easy anyways as you need a different proxy and different entries

don't feel strong about this, but I'd like to keep the API as clean as possible if we have the chance

Reply:
Neither, so will remove in favour of cleanliness